src/server/worker/api_server.go

```diff
@@ -515,10 +516,20 @@ func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLog
 		if err := createSpoutFifo(outPath); err != nil {
 			return "", fmt.Errorf("mkfifo :%v", err)
 		}
-		_, err := pachClient.InspectFile(a.pipelineInfo.Pipeline.Name, a.pipelineInfo.OutputBranch, "marker")
-		if err != nil && strings.Contains(err.Error(), "not found") {
-			if err := puller.Pull(pachClient, filepath.Join(dir, "marker"), a.pipelineInfo.Pipeline.Name, a.pipelineInfo.OutputBranch, "/marker", false, false, concurrency, nil, ""); err != nil {
-				return "", err
+		if a.pipelineInfo.Spout.Marker != "" {
+			// check if we have a marker file in the /pfs/out directory
+			_, err := pachClient.InspectFile(a.pipelineInfo.Pipeline.Name, ppsconsts.SpoutMarkerBranch, a.pipelineInfo.Spout.Marker)
+			// if not, then it might be because we need to pull it
+			if err != nil && errutil.IsNotFoundError(err) {
```

Comment:
it looks like this is still only pulling the marker file if it wasn't found. when i tried using the previous implementation, i never seemed to get my marker file back on subsequent runs. i think this is the cause.